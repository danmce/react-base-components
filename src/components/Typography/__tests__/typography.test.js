import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import { BodyText, HeadingText, TitleText } from '../'
import UxPinWrapper from '../../../wrappers/UXPinWrapper'

test('renders Typography correctly', () => {
  const bodyText = (
    <UxPinWrapper>
      <TitleText tint='#f00' bold align='center'>
        Title Text
      </TitleText>
      <HeadingText tint='#f00' bold align='center'>
        Heading Text
      </HeadingText>
      <BodyText tint='#0f0' align='right'>
        Body Text
      </BodyText>
    </UxPinWrapper>
  )

  expect(toJson(enzyme.mount(bodyText))).toMatchSnapshot('enzyme.mount')
})
