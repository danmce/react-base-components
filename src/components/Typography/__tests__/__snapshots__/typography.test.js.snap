// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Typography correctly: enzyme.mount 1`] = `
.emotion-0 {
  color: #f00;
  font-weight: bold;
  text-align: center;
  font-size: 20px;
}

.emotion-2 {
  color: #f00;
  font-weight: bold;
  text-align: center;
  font-size: 16px;
}

.emotion-4 {
  color: #0f0;
  text-align: right;
  font-size: 12px;
}

<UXPinWrapper>
  <ThemeProvider
    theme={
      Object {
        "borders": Object {
          "default": "1 solid",
          "radii": Object {
            "m": 3,
            "s": 0,
          },
        },
        "colors": Object {
          "button": Object {
            "primary": Object {
              "background": "#17384B",
              "border": "#2C546C",
              "color": "#FFFFFF",
            },
            "secondary": Object {
              "background": "transparent",
              "border": "#E9EEF2",
              "color": "#D7DFE2",
            },
          },
          "common": Object {
            "black": "#000000",
            "white": "#FFFFFF",
          },
          "error": Object {
            "contrast": "#FFFFFF",
            "fill": undefined,
          },
          "success": Object {
            "contrast": "#FFFFFF",
            "fill": undefined,
          },
          "warning": Object {
            "contrast": "#FFFFFF",
            "fill": undefined,
          },
        },
        "typography": Object {
          "family": "'Helvetica Neue', sans-serif",
          "size": Object {
            "l": 16,
            "m": 12,
            "s": 10,
            "xl": 20,
            "xs": 8,
            "xxl": 24,
            "xxxl": 34,
            "xxxxl": 50,
          },
          "weight": Object {
            "bold": 600,
            "light": 100,
            "regular": 300,
          },
        },
      }
    }
  >
    <Component
      align="center"
      bold={true}
      tint="#f00"
    >
      <StyledText
        align="center"
        bold={true}
        tag="h1"
        tint="#f00"
      >
        <WithComponent
          align="center"
          bold={true}
          tag="h1"
          tint="#f00"
        >
          <h1
            className="emotion-0 emotion-1"
          >
            Title Text
          </h1>
        </WithComponent>
      </StyledText>
    </Component>
    <Component
      align="center"
      bold={true}
      tint="#f00"
    >
      <StyledText
        align="center"
        bold={true}
        tag="h2"
        tint="#f00"
      >
        <WithComponent
          align="center"
          bold={true}
          tag="h2"
          tint="#f00"
        >
          <h2
            className="emotion-2 emotion-1"
          >
            Heading Text
          </h2>
        </WithComponent>
      </StyledText>
    </Component>
    <Component
      align="right"
      tint="#0f0"
    >
      <StyledText
        align="right"
        tag="p"
        tint="#0f0"
      >
        <WithComponent
          align="right"
          tag="p"
          tint="#0f0"
        >
          <p
            className="emotion-4 emotion-1"
          >
            Body Text
          </p>
        </WithComponent>
      </StyledText>
    </Component>
  </ThemeProvider>
</UXPinWrapper>
`;
